.werft/platform-delete-preview-environments-cron.ts

```diff
@@ -37,48 +37,102 @@ async function deletePreviewEnvironments() {
+        werft.done("Fetching preview");
     } catch (err) {
-        werft.fail("Fetching previews", err)
+        werft.fail("Fetching preview", err)
     }
-    werft.done("Fetching previews");
+
+    werft.phase("Fetching outdated branches");
+    const branches = getAllBranches();
+    const outdatedPreviews = new Set(branches
+        .filter(branch => {
+          const lastCommit = exec(`git log ${branch} --since=$(date +%Y-%m-%d -d "5 days ago")`, { silent: true })
+          return lastCommit.length < 1
+        })
+        .map(branch => expectedNamespaceFromBranch(branch.replace('origin\/','').trim())))
+
+    const expectedPreviewEnvironmentNamespaces = new Set(branches.map(branch => expectedNamespaceFromBranch(branch)));
 
     werft.phase("deleting previews")
     try {
-        const previewsToDelete = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
-        // Trigger namespace deletion in parallel
-        const promises = previewsToDelete.map(preview => wipePreviewEnvironmentAndNamespace(helmInstallName, preview, CORE_DEV_KUBECONFIG_PATH, { slice: `Deleting preview ${preview}` }));
-        // But wait for all of them to finish before (or one of them to fail) before we continue
-        await Promise.all(promises)
+        const deleteDueToMissingBranch  = previews.filter(ns => !expectedPreviewEnvironmentNamespaces.has(ns))
+        const deleteDueToNoCommitActivity  = previews.filter(ns => outdatedPreviews.has(ns))
+        const deleteDueToNoDBActivity      = previews.filter(ns => checkActivity(ns))
+        const previewsToDelete             = new Set([...deleteDueToMissingBranch, ...deleteDueToNoCommitActivity, ...deleteDueToNoDBActivity])
+
+        if (previewsToDelete.has("staging-main"))
+            previewsToDelete.delete("staging-main")
+
+        const dryRun = true
+        if (dryRun) {
+            previewsToDelete.forEach(preview => werft.log("would have deleted preview environment", preview))
+            werft.done("would have deleted preview environment")
+        }
+        else {
+            previewsToDelete.forEach(preview => werft.log("deleting preview", preview))
+            const promises = previewsToDelete.map(preview => wipePreviewEnvironmentAndNamespace(helmInstallName, preview, CORE_DEV_KUBECONFIG_PATH, { slice: `Deleting preview ${preview}` }));
+            await Promise.all(promises)
+            werft.done("deleting preview")
+        }
     } catch (err) {
-        werft.fail("deleting previews", err)
+        werft.fail("deleting preview", err)
+    }
+}
+
+
+function checkActivity(previewNS) {
+
+    const statusNS = exec(`KUBECONFIG=${CORE_DEV_KUBECONFIG_PATH} kubectl get ns ${previewNS} -o jsonpath='{.status.phase}'`, { silent: true})
+
+    if ( statusNS == "Active") {
+
+        const emptyNS = exec(`KUBECONFIG=${CORE_DEV_KUBECONFIG_PATH} kubectl get pods -n ${previewNS} -o jsonpath='{.items.*}'`, { silent: true})
+
+        if ( emptyNS.length < 1 )
```

Comment:
I'd prefer braces here too, and to use `return false` just to be explicit.

Reply:
Thanks